src/main/java/org/candlepin/model/DeletedConsumerCurator.java

```diff
@@ -94,9 +104,17 @@ public int countByConsumer(Consumer c) {
     }
 
     public int countByConsumerUuid(String uuid) {
-        return ((Long) currentSession().createCriteria(DeletedConsumer.class)
-            .add(Restrictions.eq("consumerUuid", uuid))
-            .setProjection(Projections.rowCount()).uniqueResult()).intValue();
+        if (uuid == null || uuid.isBlank()) {
+            return 0;
+        }
+
+        String query = "SELECT dc FROM DeletedConsumer dc WHERE dc.consumerUuid = :uuid";
```

Comment:
We can just do a count and not pull all the data back here:

`SELECT COUNT(dc) FROM ...`

Then the `.getResultList()` below becomes a `.getSingleResult().intValue()`

Reply:
Good catch! Updated